package/MDAnalysis/core/universe.py

```diff
@@ -124,54 +126,59 @@ def __init__(self, *args, **kwargs):
             self.atoms = None
             return
 
-        self.filename = args[0]
         coordinatefile = args[1:]
 
-        topology_format = kwargs.pop('topology_format', None)
-
-        if len(args) == 1:
-            # special hacks to treat a coordinate file as a coordinate AND topology file
-            if kwargs.get('format', None) is None:
-                kwargs['format'] = topology_format
-            elif topology_format is None:
-                topology_format = kwargs.get('format', None)
-
-            # if passed a Reader, use that
-            fmt = kwargs.get('format', None)
+        # if we're given a Topology object, we don't need to parse anything
+        if isinstance(args[0], Topology):
+            self._topology = args[0]
+            self.filename = None
+        else:
+            self.filename = args[0]
+            topology_format = kwargs.pop('topology_format', None)
```

Comment:
This was already here; the reason it counts as a change is because of the indentation into another if-else level. I see the point and will probably remove it, but as it is it is more explicit without looking up the default rules for `dict.get()`.